lib/auth/auth_with_roles.go

```diff
@@ -5455,11 +5455,29 @@ func (a *ServerWithRoles) GetAccountRecoveryCodes(ctx context.Context, req *prot
 }
 
 // GenerateCertAuthorityCRL generates an empty CRL for a CA.
+//
+// This CRL can be requested by:
+//
+//   - Windows desktop service when updating the certificate authority contents
+//     on LDAP.
+//   - Cluster administrator using "tctl auth crl --type=db" command locally
+//     on the auth server to produce revocation list used to be configured on
+//     external services such as Windows certificate store.
+//   - Remote user using "tctl auth crl --type=db" command with a remote
+//     proxy (e.g. Teleport Cloud), as long as they have permission to create
+//     a revocation list.
 func (a *ServerWithRoles) GenerateCertAuthorityCRL(ctx context.Context, caType types.CertAuthType) ([]byte, error) {
-	// Only windows_desktop_service should be requesting CRLs
-	if !a.hasBuiltinRole(types.RoleAdmin, types.RoleWindowsDesktop) {
-		return nil, trace.AccessDenied("access denied")
+	err := a.action(apidefaults.Namespace, types.KindCertAuthorityRevocationList, types.VerbCreate)
+	if err != nil {
+		if !trace.IsAccessDenied(err) {
+			return nil, trace.Wrap(err)
+		}
+
+		if !a.hasBuiltinRole(types.RoleAdmin, types.RoleWindowsDesktop) {
+			return nil, trace.AccessDenied("access denied")
+		}
 	}
+
```

Comment:
@gabrielcorado
https://github.com/gravitational/teleport/blob/master/lib/auth/auth.go#L3805 
Not related to this PR but does CRL can expire and break the Database PKINIT flow ? 
I'm wonder if we should add the ttl flag to  `tctl auth crl` command.

Reply:
Most probably, yes. That would require users to regenerate it. I think this is true if they rotate their database CA too. I can open a separate PR by adding the flag.